gremlin-go/driver/error_codes.go

```diff
@@ -0,0 +1,106 @@
+
+	// graphBinary.go errors
+	err0401WriteTypeValueUnexpectedNullError    errorCode = "E0401_GRAPH_BINARY_WRITETYPEVALUE_UNEXPECTED_NULL_ERROR"
+	err0402BytecodeWriterError                  errorCode = "E0402_GRAPH_BINARY_WRITER_BYTECODE_ERROR"
+	err0403WriteValueUnexpectedNullError        errorCode = "E0403_GRAPH_BINARY_WRITEVALUE_UNEXPECTED_NULL_ERROR"
+	err0404ReadNullTypeError                    errorCode = "E0404_GRAPH_BINARY_READ_NULLTYPE_ERROR"
+	err0405ReadValueInvalidNullInputError       errorCode = "E0405_GRAPH_BINARY_READVALUE_NULL_INPUT_ERROR"
+	err0406EnumReaderInvalidTypeError           errorCode = "E0406_GRAPH_BINARY_ENUMREADER_INVALID_TYPE_ERROR"
+	err0407GetSerializerToWriteUnknownTypeError errorCode = "E0407_GRAPH_BINARY_GETSERIALIZERTOWRITE_UNKNOWN_TYPE_ERROR"
+	err0408GetSerializerToReadUnknownTypeError  errorCode = "E0408_GRAPH_BINARY_GETSERIALIZERTOREAD_UNKNOWN_TYPE_ERROR"
+
+	// protocol.go errors
+	err0501ResponseHandlerResultSetNotCreatedError errorCode = "E0501_PROTOCOL_RESPONSEHANDLER_NO_RESULTSET_ON_DATA_RECEIVE"
+	err0502ResponseHandlerReadLoopError            errorCode = "E0502_PROTOCOL_RESPONSEHANDLER_READ_LOOP_ERROR"
+	err0503ResponseHandlerAuthError                errorCode = "E0503_PROTOCOL_RESPONSEHANDLER_AUTH_ERROR"
+
+	// result.go errors
+	err0601ResultNotVertexError         errorCode = "E0601_RESULT_NOT_VERTEX_ERROR"
+	err0602ResultNotEdgeError           errorCode = "E0602_RESULT_NOT_EDGE_ERROR"
+	err0603ResultNotElementError        errorCode = "E0603_RESULT_NOT_ELEMENT_ERROR"
+	err0604ResultNotPathError           errorCode = "E0604_RESULT_NOT_PATH_ERROR"
+	err0605ResultNotPropertyError       errorCode = "E0605_RESULT_NOT_PROPERTY_ERROR"
+	err0606ResultNotVertexPropertyError errorCode = "E0606_RESULT_NOT_VERTEX_PROPERTY_ERROR"
+	err0607ResultNotTraverserError      errorCode = "E0607_RESULT_NOT_TRAVERSER_ERROR"
+	err0608ResultNotSliceError          errorCode = "E0608_RESULT_NOT_SLICE_ERROR"
+
+	// serializer.go errors
+	err0701ReadMapNullKeyError          errorCode = "E0701_SERIALIZER_READMAP_NULL_KEY_ERROR"
+	err0703ReadMapNonStringKeyError     errorCode = "E0703_SERIALIZER_READMAP_NON_STRING_KEY_ERROR"
+	err0704ConvertArgsNoSerializerError errorCode = "E0704_SERIALIZER_CONVERTARGS_NO_SERIALIZER_ERROR"
+
+	// transporterFactory.go errors
+	err0801GetTransportLayerNoTypeError errorCode = "E0801_TRANSPORTERFACTORY_GETTRANSPORTLAYER_NO_TYPE_ERROR"
+
+	// traversal.go errors
+	err0901ToListAnonTraversalError  errorCode = "E0901_TRAVERSAL_TOLIST_ANON_TRAVERSAL_ERROR"
+	err0902IterateAnonTraversalError errorCode = "E0902_TRAVERSAL_ITERATE_ANON_TRAVERSAL_ERROR"
+	err0903NextNoResultsLeftError    errorCode = "E0903_TRAVERSAL_NEXT_NO_RESULTS_LEFT_ERROR"
+
+	// bytecode.go errors
+	err1001ConvertArgumentChildTraversalNotFromAnonError errorCode = "E1001_BYTECODE_CHILD_T_NOT_ANON_ERROR"
+)
+
+func NewError(errorCode errorCode, args ...interface{}) error {
+	localizer := getLocalizer(language.English)
```

Comment:
Loading a json file for each error is too expensive. It would be nice to make a cache

Reply:
How can I go about making a cached system?

Reply:
simple: store localizer in variable and reuse each time
something like https://go.dev/play/p/C1Sz1Ft1I20?v=goprev